pkg/flow/internal/controller/value_cache.go

```diff
@@ -165,36 +135,5 @@ func (vc *valueCache) buildValue(from []ComponentID, offset int) cty.Value {
 	for label, ids := range componentsByLabel {
 		attrs[label] = vc.buildValue(ids, offset+1)
 	}
-
-	return cty.ObjectVal(attrs)
-}
-
-// mergeComponentValues merges a component's config and exports. mergeState
-// panics if a key exits in both inputs and store or if neither argument is an
-// object.
-func mergeComponentValues(args, exports cty.Value) cty.Value {
-	if !args.Type().IsObjectType() {
-		panic("component arguments must be object type")
-	}
-	if !exports.Type().IsObjectType() {
-		panic("component exports must be object type")
-	}
-
-	var (
-		inputMap = args.AsValueMap()
-		stateMap = exports.AsValueMap()
-	)
-
-	mergedMap := make(map[string]cty.Value, len(inputMap)+len(stateMap))
-	for key, value := range inputMap {
-		mergedMap[key] = value
-	}
-	for key, value := range stateMap {
-		if _, exist := mergedMap[key]; exist {
-			panic(fmt.Sprintf("component exports overrides arguments key %s", key))
-		}
-		mergedMap[key] = value
-	}
-
-	return cty.ObjectVal(mergedMap)
```

Comment:
This was removed after a discussion with @mattdurham; it's too confusing to explain to users how arguments get converted into values, but much easier to explain exports. For now, since we can't think of a use case where a user would need to reference an argument, only the exports can be referenced.